File: unit-test-practice-2/src/test/java/dojo/ComparableVersionTest.java
package dojo;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.testng.annotations.Test;

public class ComparableVersionTest {

	@Test
	public void should_do_something() throws Exception {
		int result = 12;
		// asszertálás példák:
		assertThat(result).isEqualTo(12);
		assertThat(result).isPositive();
		assertThat(result).isNotNegative();
		assertThat(result).isNotZero();
		// asszertálás kivételre:
		assertThatThrownBy(() -> {
			throw new NullPointerException();
		}).isInstanceOf(NullPointerException.class);
	}
	
}

File: unit-test-practice-2/src/test/kotlin/dojo/ComparableVersionTest.kt
package dojo

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class ComparableVersionTest {

    @Test
    fun `should do something`() {
        val result = 12
        // asszertálás példák:
        assertThat(result).isEqualTo(12)
        assertThat(result).isPositive()
        assertThat(result).isNotNegative()
        assertThat(result).isNotZero()
        // asszertálás kivételre:
        assertThatThrownBy {
            throw NullPointerException()
        }.isInstanceOf(NullPointerException::class.java)
    }
}